kubernetes-client/src/main/java/io/fabric8/kubernetes/client/CustomResourceDoneable.java

```diff
@@ -0,0 +1,37 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.kubernetes.client;
+
+import io.fabric8.kubernetes.api.builder.Function;
+import io.fabric8.kubernetes.api.model.Doneable;
+import io.fabric8.kubernetes.api.model.HasMetadata;
+
+/**
+ */
+public class CustomResourceDoneable<T extends HasMetadata> implements Doneable<T> {
```

Comment:
I am wondering if we can (and also if it makes sense to) make things more strict here and have T extends CustomResourceSupport (or whatever we end up naming this class).

Reply:
yeah - sounds good to me. I guess folks can always opt out of the CustomResource / CustomResourceDoneable classes anyway if they want to do things themselves